otherlibs/unix/termios.c

```diff
@@ -314,12 +313,11 @@ static int when_flag_table[] = {
 
 CAMLprim value caml_unix_tcsetattr(value fd, value when, value arg)
 {
-  if (tcgetattr(Int_val(fd), &terminal_status) == -1)
-    caml_uerror("tcsetattr", Nothing);
-  decode_terminal_status(&Field(arg, 0));
```

Comment:
I did not quite understand why `tcsetattr` calls `tcgetattr`, for me it does not change the result so I removed it, but I just want to mention this in case I'm missing something.

Reply:
Well removing this does change the semantics. As of now, `tcsetattr` updates the fields of its `terminal_io` argument. I don't quite see when this can be useful. But if existing code relies on this, maybe we don't want to change it.

Reply:
In general, `struct termios` contains additional, OS-specific fields and bits beyond the standard ones that are mapped to the `Unix.terminal_io` OCaml type.  It's better not to change these additional fields.

Reply:
Understood, I put the call to `tcgetattr` back.